src/microbe_stage/editor/TolerancesEditorSubComponent.cs

```diff
@@ -709,12 +759,51 @@ private void OnUVResistanceSliderChanged(float value)
 
         if (!TriggerChangeIfPossible())
         {
-            uvResistanceSlider.Value = CurrentTolerances.UVResistance;
+            var extremeResistance = CalculateSliderExtremeValue(Constants.TOLERANCE_CHANGE_MP_PER_UV,
+                value, CurrentTolerances.UVResistance, true);
+
+            reusableTolerances.UVResistance = extremeResistance;
+            uvResistanceSlider.Value = extremeResistance;
+
+            // Attempt the previous rollback if failed again.
+            if (!TriggerChangeIfPossible())
+            {
+                uvResistanceSlider.Value = CurrentTolerances.UVResistance;
+            }
         }
 
         automaticallyChanging = false;
     }
 
+    /// <summary>
+    ///   Calculates the extremeties of a slider movement when cost is the limiting factor.
+    /// </summary>
+    /// <param name="cpa">Cost per action. How much MP does changing this value cost?</param>
+    /// <param name="sliderValue">What the slider value currently is at</param>
+    /// <param name="originalValue">What the slider value was originally at</param>
+    /// <param name="isPercentage">
+    ///   If the value being calculated is a percentage. This is used
+    ///   for determining the rounding for the amount we can change by.
+    /// </param>
+    /// <returns>The value the slider and parameter should now be at.</returns>
+    private float CalculateSliderExtremeValue(double cpa, float sliderValue, float originalValue,
+        bool isPercentage = false)
+    {
+        var pointsLeft = Editor.MutationPoints;
+
+        // If a percentage, round to the nearest 0.05
+        var numActions = isPercentage ? Math.Floor(pointsLeft * 20 / cpa) / 20 : Math.Floor(pointsLeft / cpa);
```

Comment:
I think this is not a safe assumption that things must be rounded to nearest 0.05. Someone could edit the GUI. So instead I think what you need to do is pass in the step size to this method and have the callers read the values from the GUI slider step size.